src/java/org/apache/cassandra/io/sstable/format/big/BigTableBlockWriter.java

```diff
@@ -0,0 +1,221 @@
+
+        assert supportedComponents.containsAll(components) : String.format("Unsupported streaming component detected %s",
+                                                                           new HashSet(components).removeAll(supportedComponents));
+
+        for (Component c : components)
+            componentWriters.put(c.type, makeWriter(descriptor, c, writerOption));
+    }
+
+    private static SequentialWriter makeWriter(Descriptor descriptor, Component component, SequentialWriterOption writerOption)
+    {
+        return new SequentialWriter(new File(descriptor.filenameFor(component)), writerOption, false);
+    }
+
+    private void write(DataInputPlus in, long size, SequentialWriter out) throws FSWriteError
+    {
+        final int BUFFER_SIZE = 1 * 1024 * 1024;
+        long bytesRead = 0;
+        byte[] buff = new byte[BUFFER_SIZE];
+        try
+        {
+            while (bytesRead < size)
+            {
+                int toRead = (int) Math.min(size - bytesRead, BUFFER_SIZE);
+                in.readFully(buff, 0, toRead);
+                int count = Math.min(toRead, BUFFER_SIZE);
+                out.write(buff, 0, count);
+                bytesRead += count;
+            }
+            out.sync();
+        }
+        catch (IOException e)
+        {
+            throw new FSWriteError(e, out.getPath());
+        }
+    }
+
+    @Override
+    public boolean append(UnfilteredRowIterator partition)
+    {
+        throw new UnsupportedOperationException("Operation not supported by BigTableBlockWriter");
+    }
+
+    @Override
+    public Collection<SSTableReader> finish(long repairedAt, long maxDataAge, boolean openResult)
+    {
+        return finish(openResult);
+    }
+
+    @Override
+    public Collection<SSTableReader> finish(boolean openResult)
+    {
+        setOpenResult(openResult);
+        return finished();
+    }
+
+    @Override
+    public Collection<SSTableReader> finished()
+    {
+        if (finalReader == null)
+            finalReader = SSTableReader.open(descriptor,
+                                             components,
+                                             metadata);
+
+        return ImmutableList.of(finalReader);
+    }
+
+    @Override
+    public SSTableMultiWriter setOpenResult(boolean openResult)
+    {
+        return null;
+    }
+
+    @Override
+    public long getFilePointer()
+    {
+        return 0;
+    }
+
+    @Override
+    public TableId getTableId()
+    {
+        return metadata.id;
+    }
+
+    @Override
+    public Throwable commit(Throwable accumulate)
+    {
+        for (SequentialWriter writer : componentWriters.values())
+            writer.commit(accumulate);
+
+        return accumulate;
+    }
+
+    @Override
+    public Throwable abort(Throwable accumulate)
+    {
+        for (SequentialWriter writer : componentWriters.values())
+            writer.abort(accumulate);
+
+        return accumulate;
```

Comment:
We shouldn't be ignoring the result of writer.abort(), and should return it instead of accumulate.

Reply:
Cherry picked.